cypress/integration/Tools/TestUiViewAndLogs.spec.js

```diff
@@ -0,0 +1,84 @@
+/**
+* Copyright (c) Microsoft Corporation. All rights reserved.  
+ * Licensed under the MIT License.
+*/
+
+// -----------------------------------------------------------------------------
+// The Purpose of this Test Spec File is to test the Persistent Logging features
+// and to demonstrate how to skip dependant tests when their predicessor fails.
+// -----------------------------------------------------------------------------
+
+const helpers = require('../../support/Helpers')
+
+// The difference between the 1st and 2nd is where we place the call to SkipRemainingTestsOfSuiteIfFailed()
+describe('1st Test Suite Root', () => {
+  afterEach(helpers.SkipRemainingTestsOfSuiteIfFailed)
```

Comment:
Placing the `Skip...` here causes all remaining tests of the suite to be skipped if there is a failure.

Reply:
Would need to make the support the case where people want to write other things in the `afterEach`

Like:
```
afterEach(() => {
  someCleanUp()
  otherCustomCode()
  helpers.SkipRemainingTestsOfSuiteIfFailed()
})
```

I imagine you'll have to pass in stuff from the outer `this`.  Also we want to support short hand functions `() => { ... }`.  I think cypress has alternatives using `this`

Reply:
You bring up some interesting points:

1. JavaScript does not give you the "this" object in the same way when you use arrow function as when you use named or unnamed functions. Thus whenever you use the `SkipRemainingTestsOfSuiteIfFailed` function you cannot use arrow functions (lambdas). So you can do what you want if you change to this:

```suggestion
afterEach(function() {
  someCleanUp()
  otherCustomCode()
  helpers.SkipRemainingTestsOfSuiteIfFailed()
})
```
2. Under what circumstances would you need to do cleanup? It is highly recommended to avoid cleanup in the test code since we cannot guarantee that it will happen. And if we depend on it happening in future tests then we have a big mess.

3. You can also repeat all of the hooks since they add them to some sort of event queue. So if necessary you could have one afterEach just as I specified it, and then a second one to deal with your other needs...I found this handy when I had to make sure all of the Cypress `cy` calls were completed before moving on to the next bits of code that needed to be executed after the `cy` functions finished.

Reply:
I did not understand this part of your comment...
> I think cypress has alternatives using this

Reply:
> JavaScript does not give you the "this" object in the same way when you use arrow function

Yes, that's why we shouldn't rely on `this` since it's not always that you think.  We don't want to have to remember this special case not being able to use lambdas when write an `afterEach`

> Under what circumstances would you need to do cleanup?

Hmm, i had suspicion that example would confuse you.  Only chose it because that's what most people do in `afterEach`.  The code can be anything.  It's just a function. For example in your code you do logging.

Reply:
I looked for an alternative in the Cypress docs to the `this` but found none.

I just added comments to remind them AND an exception if they call it wrongly so as to remind them of the correct way to call it.

> Test Code Error: Cannot use arrow/lambda function to call the SkipRemainingTestsOfSuiteIfFailed() function.
>

Reply:
So did you end up making this work when it is mixed with other functions? E.g. more than skipping tests in the `afterEach`.

Otherwise "can't use lambdas" is only 1 of the special cases and the other is "you also can't do anything else in this function".

Reply:
The answer is yes, it does work when mixed with other function calls. I had already addressed this in my first reply #1. Not only that but you can also supply multiple calls to `afterEach` with each one containing different functionality, both will be executed in the order that `afterEach` was called.